cmd/create/machinepool/nodepool.go

```diff
@@ -31,9 +32,29 @@ func addNodePool(cmd *cobra.Command, clusterKey string, cluster *cmv1.Cluster, r
 		os.Exit(1)
 	}
 
-	// Hosted clusters create identifiers for NodePools, users don't interact directly with these resources
-	if cmd.Flags().Changed("name") {
-		r.Reporter.Errorf("Setting the `name` is not supported for hosted clusters")
+	// Machine pool name:
+	name := strings.Trim(args.name, " \t")
+	if name == "" && !interactive.Enabled() {
+		interactive.Enable()
+		r.Reporter.Infof("Enabling interactive mode")
+	}
+	if name == "" || interactive.Enabled() {
+		name, err = interactive.GetString(interactive.Input{
+			Question: "Machine pool name",
+			Default:  name,
+			Required: true,
```

Comment:
Is there a benefit to making this required vs optional? We could easily generate `$cluster.name-$randomhash` as a default instead of forcing the user.

Reply:
No idea, at least the requeriment was to make the NodePools to operate as the classic machinepool is working today.

https://github.com/openshift/rosa/blob/master/cmd/create/machinepool/machinepool.go#L69

Yeah, I like that suggestion; perhaps when consolidated, we could propose it for all cases? (classic rosa and hosted clusters)